symlink-check/symlink-check

```diff
@@ -0,0 +1,205 @@
+#!/bin/sh
+# shellcheck disable=SC2317,SC2166,SC3043,SC2162,SC2086
+set -f
+
+PROG="symlink-check"
+
+info() {
+	echo "INFO[$PROG]:" "$@"
+}
+
+error() {
+	echo "ERROR[$PROG]:" "$@"
+	exit 1
+}
+
+fail() {
+	FAIL_MESSAGES="${FAIL_MESSAGES}FAIL[$PROG]: $*
+"
+	FAILS=$((FAILS+1))
+}
+
+pass() {
+	echo "PASS[$PROG]:" "$@"
+	PASSES=$((PASSES+1))
+}
+
+show_help() {
+	cat << EOF
+Usage: $PROG [OPTIONS]
+
+Tool to check for broken/dangling symlinks in the filesystem.
+
+Options:
+  -h, --help                    Show this help message and exit
+  --paths=PATH, --paths PATH    Specify paths to check (default: /)
+  --packages=PKG, --packages PKG
+                               Specify packages to check
+  --verbose=BOOL, --verbose BOOL
+                               Enable verbose output (true or false)
+
+Examples:
+  $PROG --paths=/usr/bin
+  $PROG --packages=bash
+  $PROG --paths=/usr/bin --verbose=true
+EOF
+	exit 0
+}
+
+paths=""
+packages=""
+VERBOSE=false
+
+while [ $# -ne 0 ]; do
+	case "$1" in
+		-h|--help) show_help;;
+		--paths=*) paths="${paths} ${1#*=}";;
+		--paths) paths="${paths} $2"; shift;;
+		--packages=*)
+			if [ "${1#*=}" = "none" ]; then
+				packages=""
+			else
+				packages="${packages} ${1#*=}"
+			fi
+		;;
+		--packages)
+			if [ "$2" = "none" ]; then
+				packages=""
+			else
+				packages="${packages} $2"
+			fi
+			shift
+		;;
+		--verbose=*) VERBOSE=${1#*=};;
+		--verbose) VERBOSE=$2; shift;;
+		--*) error "Unknown argument '$1'";;
+	esac
+	shift
+done
+
+paths=${paths# }
+packages=${packages# }
+
+case "$VERBOSE" in
+	true|false) :;;
+	*) error "--verbose must be 'true' or 'false'. found '$VERBOSE'";;
+esac
+
+[ -n "${paths}${packages}" ] || paths="/"
+
+export LANG=C
```

Comment:
Can we move the main logics into a `main` function? Probably merging this with block 178:205